health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/goals/ExerciseGoalsScreen.kt

```diff
@@ -0,0 +1,228 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.wear.compose.material.Chip
+import androidx.wear.compose.material.CompactButton
+import androidx.wear.compose.material.CompactChip
+import androidx.wear.compose.material.Icon
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.ui.tooling.preview.WearPreviewDevices
+import com.example.exercisesamplecompose.R
+import com.example.exercisesamplecompose.presentation.goals.ExerciseGoalsViewModel
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.compose.layout.ScalingLazyColumn
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults.ItemType
+import com.google.android.horologist.compose.layout.ScreenScaffold
+import com.google.android.horologist.compose.layout.rememberResponsiveColumnState
+
+
+@Composable
+fun ExerciseGoalsRoute(
+    onSet: () -> Unit,
+) {
+    val viewModel: ExerciseGoalsViewModel = hiltViewModel()
+
+    ExerciseGoalsScreen(onSet, setGoals = { distance: Double, duration: Long ->
+        viewModel.setGoals(distance, duration)
+    })
+}
+
+@Composable
+fun ExerciseGoalsScreen(
+    onSet: () -> Unit = {}, setGoals: (Double, Long) -> Unit
+
+) {
+
+    // Clear up screen real-estate while toggling goal values
+    var showDistanceRow by remember { mutableStateOf(false) }
+    var showDurationRow by remember { mutableStateOf(false) }
+
+    // Default values for goals setting
+    var distanceValue by remember { mutableDoubleStateOf(0.0) }
+    var durationValue by remember { mutableDoubleStateOf(0.0) }
+
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = ScalingLazyColumnDefaults.padding(
+            first = ItemType.Text,
+            last = ItemType.CompactChip,
+        )
+    )
+    ScreenScaffold(scrollState = columnState) {
+        ScalingLazyColumn(
+            modifier = Modifier.fillMaxSize(),
+            columnState = columnState,
+        ) {
+            if (!showDistanceRow) {
+                item {
+                    Chip(onClick = { showDistanceRow = !showDistanceRow },
+                        label = { Text(stringResource(R.string.add_distance)) })
+                }
+            }
+            if (showDistanceRow) {
+                item {
+                    DistanceGoalRow(distanceValue) { newValue ->
+                        distanceValue = newValue
+                    }
+                }
+            }
+            if (!showDurationRow) {
```

Comment:
I wonder whether this should be handled via expanded items and InlineSlider?

This is a quick hacked up version

https://gist.github.com/yschimke/40f357e7f2f5078cfdb70ca8cbd1bdca